src/Common/src/Interop/Unix/libcrypto/Interop.Initialization.cs

```diff
@@ -21,8 +18,15 @@ static libcrypto()
                 throw new CryptographicException();
             }
 
+            // Load the SHA-2 hash algorithms, and anything else not in the default
+            // support set.
+            OPENSSL_add_all_algorithms_conf();
+
             // Ensure that the error message table is loaded.
-            Interop.libcrypto.ERR_load_crypto_strings();
+            ERR_load_crypto_strings();
         }
```

Comment:
In the PR description you mentioned:

> * Differs from Windows behaviors in cases of multiple errors. Windows reports every error for every cert in the chain, OpenSSL reports only the first error it finds.

But looking at manual: http://www.openssl.org/docs/crypto/ERR_load_crypto_strings.html, http://wiki.openssl.org/index.php/Manual:ERR_print_errors%283%29 and PHP implementation: http://php.net/manual/en/function.openssl-error-string.php, it appears that all the errors are recorded as a stack and can be read by iterating over the function, something like:

```
// pseudo-code
for(auto i : ERR_load_crypto_strings()) { .. }
```

?

@khouzam, do you know of better way of reading "all" crypt errors in OpenSSL?